### ISSUE TITLE
BrainTreeFragment ConcurrentModificationException please merge pull request

### ISSUE THREAD

**Issue by skauss** [braintree/braintree_android#283]
We have in production a lot of "Fatal Exception" (23 in 7 days):
```
 java.util.ConcurrentModificationException
    at java.util.ArrayDeque$DeqIterator.next + 625(ArrayDeque.java:625)
    at java.util.AbstractCollection.addAll + 343(AbstractCollection.java:343)
    at java.util.ArrayDeque.<init> + 197(ArrayDeque.java:197)
    at com.braintreepayments.api.BraintreeFragment.flushCallbacks + 726(BraintreeFragment.java:726)
   at com.braintreepayments.api.BraintreeFragment.onResume + 249(BraintreeFragment.java:249)
```
The only part where mCallbackQueue is not syncronized is postOrQueueCallback.
I suppose while make a copy of mCallbackQueue in flushCallbacks (line 726) an other thread call postOrQueueCallback

Please submit my merge request and dispose a new version.

https://github.com/braintree/braintree_android/pull/282/files

Regards
Stephan

**Comment by sestevens** [comment]
Hi @skauss, apologies for the delay! We've taken a look at your pull request, and it looks like a reasonable change. We will try to do a final review soon and move it along in the process.

**Comment by sestevens** [comment]
This has been merged and will go out in our next release (mostly likely sometime in the next few weeks).

**Comment by skauss** [comment]
Hi @sestevens 
please let me know the release date. We still have many crashes.

**Comment by sestevens** [comment]
@skauss This was released at the end of the day yesterday in version 3.7.2.

**Comment by skauss** [comment]
@sestevens 
The is the **com.braintreepayments.api:braintree** Version Number but I need the **drop-in** version number.
On Page https://developers.braintreepayments.com/start/hello-client/android/v3 the Version is still implementation 'com.braintreepayments.api:drop-in:4.4.1'

The latest version on https://mvnrepository.com/artifact/com.braintreepayments.api/drop-in

Is also 4.4.1

Please let me know where I find the latest  drop-in library
Regards
Stephan

**Comment by sestevens** [comment]
Oh, I see! I will check with our team to see how quickly we can get a Drop-in release out.

**Comment by scannillo** [comment]
Hey @skauss. We released version of [Android Drop-in SDK 4.5.0](https://github.com/braintree/braintree-android-drop-in/releases/tag/4.5.0). This will rely on Braintree Android version 3.7.2, and has the fix you're looking for!
